sstables/sstables.cc

```diff
@@ -927,6 +927,12 @@ void sstable::generate_toc(compressor_ptr c, double filter_fp_chance) {
     _recognized_components.insert(component_type::Scylla);
 }
 
+file_writer sstable::make_component_file_writer(component_type c, file_output_stream_options options, open_flags oflags)
+{
+    auto f = new_sstable_component_file(_write_error_handler, c, oflags).get0();
+    return file_writer(std::move(f), std::move(options));
```

Comment:
What happens if file_writer's constructor throws? We don't close f.

Reply:
good catch.  I'll add handling for that.